labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -37,15 +45,36 @@ object RecursionPatternMatchingExercise {
    * List(1,1,2,3,1,1) -> List(1,1), List(2), List(3), List(1,1)
    */
   def groupConsecutive[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+    in match {
+      case List() => List()
+      case a :: rest =>
+        groupConsecutive(rest) match {
+          case (b::rrest) :: groups if b == a => (a::b::rrest) :: groups
+          case groups => List(a) :: groups
+        }
+    }
   }
 
+  
   /**
    * Group Equal values
    * List(1,1,2,3,1,1) -> List(1,1,1,1), List(2), List(3)
    */
   def groupEquals[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+    // groupEquals never produces empty groups
+    groupEqualsHelp(in).values.toList sortBy {case a::_ => in indexOf a}
+  }
```

Comment:
a helper isn't really needed as you can use partition to get all of one value, then move onto the next.